homeassistant/components/tfl/config_flow.py

```diff
@@ -0,0 +1,284 @@
+                self.options[CONF_STOP_POINTS] = []
+                return await self.async_step_stop_point()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_stop_point(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the stop point step."""
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+        if user_input is not None:
+            app_key = self.data[CONF_API_APP_KEY]
+            errors, description_placeholders = await validate_stop_points(
+                self.hass, app_key, user_input[CONF_STOP_POINTS]
+            )
+
+            if not errors:
+                # User input is valid, save the stop point
+                self.options[CONF_STOP_POINTS].extend(user_input[CONF_STOP_POINTS])
+
+                # Create the entry
+                return self.async_create_entry(
+                    title="Transport for London", data=self.data, options=self.options
+                )
+
+        return self.async_show_form(
+            step_id="stop_point",
+            data_schema=STEP_STOP_POINT_DATA_SCHEMA,
+            errors=errors,
+            description_placeholders=description_placeholders,
+        )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Add reconfigure step to allow to reconfigure a config entry."""
+        errors: dict[str, str] = {}
+        config_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        assert config_entry is not None
+        if user_input is not None:
+            try:
+                await validate_app_key(self.hass, user_input[CONF_API_APP_KEY])
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                # Input is valid, set data
+                self.data[CONF_API_APP_KEY] = user_input[CONF_API_APP_KEY]
+                self.hass.config_entries.async_update_entry(
+                    config_entry, data=self.data
+                )
+                await self.hass.config_entries.async_reload(self.context["entry_id"])
+                return self.async_abort(reason="reconfigure_successful")
+
+        return self.async_show_form(
+            step_id="reconfigure",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_API_APP_KEY, default=config_entry.data[CONF_API_APP_KEY]
+                    ): cv.string,
+                }
+            ),
```

Comment:
Instead of this, use a constant schema and call `add_suggested_values_to_schema`:

```py
data_schema=self.add_suggested_values_to_schema(
    RECONFIGURE_SCHEMA, config_entry.data
)
```